karma.conf.js

```diff
@@ -1,80 +1,97 @@
+      'test/non_default.js',
+      'src/plugins/navigation/navigation_tests.js',
+      'test/core_tests_presentation.html',
+      'test/non_default.html',
+      'js/impress.js',
+      'node_modules/syn/dist/global/syn.js'
+    ],
+
+    proxies: {
+    "/test/": "/base/test/",
+    "/js/": "/base/js/",
+    "/node_modules/": "/base/node_modules/"
+  },
+    
+    client: {
+      clearContext: false,
+      qunit: {
+        showUI: true
+      }
+    },
+
+    // list of files / patterns to exclude
+    exclude: [
     ],
 
-    // List of files to exclude
-    exclude: [],
 
-    // Test results reporter to use
-    // possible values: 'dots', 'progress', 'junit', 'growl', 'coverage'
-    reporters: [ "progress" ],
+    // preprocess matching files before serving them to the browser
+    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+    preprocessors: {
+    },
+
+
+    // test results reporter to use
+    // possible values: 'dots', 'progress'
+    // available reporters: https://npmjs.org/browse/keyword/karma-reporter
+    reporters: ['progress'],
+
 
-    // Web server port
+    // web server port
     port: 9876,
 
-    // Enable / disable colors in the output (reporters and logs)
+
+    // enable / disable colors in the output (reporters and logs)
     colors: true,
 
-    // Level of logging
-    // possible values: config.LOG_DISABLE || config.LOG_ERROR ||
-    // config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
-    logLevel: config.LOG_DEBUG,
 
-    // Enable / disable watching file and executing tests whenever any file changes
+    // level of logging
+    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+    logLevel: config.LOG_INFO,
+
+
+    // enable / disable watching file and executing tests whenever any file changes
     autoWatch: true,
 
-    // Start these browsers, currently available:
-    // - Chrome
-    // - ChromeCanary
-    // - Firefox
-    // - Opera (has to be installed with `npm install karma-opera-launcher`)
-    // - Safari (only Mac; has to be installed with `npm install karma-safari-launcher`)
-    // - PhantomJS
-    // - IE (only Windows; has to be installed with `npm install karma-ie-launcher`)
-    //browsers: [ "Chrome" ],
-    //browsers: [ "Firefox" ],
-    browsers: [ "Chrome", "Firefox" ],
 
-    client: {
-      clearContext: false,
-      qunit: {
-        showUI: true,
-        testTimeout: 120*1000
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['FirefoxHeadless', 'Chrome_without_security'],
+
+    customLaunchers: {
+      Chrome_without_security: {
+        base: 'ChromeHeadless',
+        flags: ['--no-sandbox']
       }
     },
 
-    // If browser does not capture, or produce output, in given timeout [ms], kill it
-    captureTimeout: 60*1000,
-    browserNoActivityTimeout: 60*1000,
-    
```

Comment:
Why are the timeouts no longer needed? I can see the tests passing, just want to learn and understand.

Reply:
I just started with the official template. Therefore lots of things are reordered or removed in that file.

Reply:
btw there are also sane defaults (http://karma-runner.github.io/3.0/config/configuration-file.html)